uvtools/dspec.py

```diff
@@ -1,78 +1,160 @@
+
+    Returns:
+        d_mdl: best fit low-pass filter components (CLEAN model) in real space
+        d_res: best fit high-pass filter components (CLEAN residual) in real space
+        info: dictionary (1D case) or list of dictionaries (2D case) with CLEAN metadata
+    '''
     nchan = data.shape[-1]
     window = aipy.dsp.gen_window(nchan, window=window)
     _d = np.fft.ifft(data * wgts * window, axis=-1)
     _w = np.fft.ifft(wgts * wgts * window, axis=-1)
-    uthresh,lthresh = wedge_width(bl_len, sdf, nchan, standoff=standoff, horizon=horizon)
+    uthresh,lthresh = calc_width(filter_size, real_delta, nchan)
     area = np.ones(nchan, dtype=np.int); area[uthresh:lthresh] = 0
     if data.ndim == 1:
         _d_cl, info = aipy.deconv.clean(_d, _w, area=area, tol=tol, stop_if_div=False, maxiter=maxiter)
-        d_mdl = np.fft.fft(_d_cl)# + info['res'])
+        d_mdl = np.fft.fft(_d_cl)
+        del info['res']
     elif data.ndim == 2:
+        info = []
         d_mdl = np.empty_like(data)
         for i in xrange(data.shape[0]):
-            if _w[i,0] < skip_wgt: d_mdl[i] = 0 # skip highly flagged (slow) integrations
+            if _w[i,0] < skip_wgt: 
+                d_mdl[i] = 0 # skip highly flagged (slow) integrations
             else:
-                _d_cl, info = aipy.deconv.clean(_d[i], _w[i], area=area, tol=tol, 
-                        stop_if_div=False, maxiter=maxiter)
-                d_mdl[i] = np.fft.fft(_d_cl)# + info['res'])
-                # XXX info overwritten every i
-    else: raise ValueError('data must be a 1D or 2D array')
+                _d_cl, info_here = aipy.deconv.clean(_d[i], _w[i], area=area, tol=tol, stop_if_div=False, maxiter=maxiter)
+                d_mdl[i] = np.fft.fft(_d_cl)
+                del info_here['res']
+                info.append(info_here)
+    else: 
+        raise ValueError('data must be a 1D or 2D array')
     d_res = data - d_mdl * wgts
+
     return d_mdl, d_res, info
 
-def delay_filter_aa(aa, data, wgts, i, j, sdf, phs2lst=False, jds=None, 
-        skip_wgt=0.5, lst_res=binning.DEFAULT_LST_RES, standoff=0., horizon=1., 
-        tol=1e-4, window='none', maxiter=100):
-    '''Use information from AntennaArray object to delay filter data, with the
-    option to phase data to an lst bin first.  Arguments are the same as for
-    delay_filter and binning.phs2lstbin.  Returns mdl, residual, and info
-    in the frequency domain.'''
-    if phs2lst:
-        data = binning.phs2lstbin(data, aa, i, j, jds=jds, lst_res=lst_res)
-    bl = aa.get_baseline(i,j)
-    return delay_filter(data, wgts, np.linalg.norm(bl), sdf, 
-            standoff=standoff, horizon=horizon, tol=tol, window=window, 
-            skip_wgt=skip_wgt, maxiter=maxiter)
+
+    
+def delay_filter(data, wgts, bl_len, sdf, standoff=0., horizon=1., tol=1e-4, 
+        window='none', skip_wgt=0.5, maxiter=100):
+    '''Apply a wideband delay filter to data. Variable names preserved for 
```

Comment:
The function names have changed, so do we need to keep the old variable names? If backwards compatibility is important, it might be worth writing a thin wrapper around the new functions that provides functions with the old names and keywords.

Reply:
delay_filter's function name actually has not changed... though github's diff makes that a bit hard to see.